notary/notary.spec

```diff
@@ -0,0 +1,310 @@
+%global with_devel 1
+#FIXME? global with_bundled 0
+%global with_bundled 1
+%global with_debug 1
+%global with_check 1
+%global with_unit_test 1
+%else
+%global with_devel 0
+#FIXME? global with_bundled 0
+%global with_bundled 1
+%global with_debug 1
+%global with_check 1
+%global with_unit_test 0
+%endif
+
+%if 0%{?with_debug}
+%global _dwz_low_mem_die_limit 0
+%else
+%global debug_package   %{nil}
+%endif
+
+%global provider        github
+%global provider_tld    com
+%global project         docker
+%global repo            notary
+# https://github.com/docker/notary
+%global provider_prefix %{provider}.%{provider_tld}/%{project}/%{repo}
+%global import_path     %{provider_prefix}
+%global commit          8a5c8c04955ba32f7e1fcd66571c6a8988e251b0
+%global shortcommit     %(c=%{commit}; echo ${c:0:7})
+
+Name:           notary
+Version:        0.1
+Release:        1.20160219git%{shortcommit}%{?dist}
+Summary:        A server and client for running and interacting with trusted collections
+License:        ASL 2.0
+URL:            https://%{provider_prefix}
+Source0:        https://%{provider_prefix}/archive/%{commit}/%{repo}-%{shortcommit}.tar.gz
+Source1:        notary-server.service
+Source2:        notary-signer.service
+# Edits example config files to make it easier to start with Notary, but the
+# edits also (intentionally) prevent running with unmodified configuration;
+# OTOH upstream wants the unmodified configuration to work in docker-compose
+# setups. So, not sent upstream.
+Patch0:         notary-config.patch
+
+
+# e.g. el6 has ppc64 arch without gcc-go, so EA tag is required
+ExclusiveArch:  %{?go_arches:%{go_arches}}%{!?go_arches:%{ix86} x86_64 %{arm}}
+# If go_compiler is not set to 1, there is no virtual provide. Use golang instead.
+BuildRequires:  %{?go_compiler:compiler(go-compiler)}%{!?go_compiler:golang}
+BuildRequires: libtool-ltdl-devel systemd
+Requires(post): systemd
+Requires(preun): systemd
+Requires(postun): systemd
+
+%description
+%{summary}
+
+%post
+%systemd_post notary-server.service notary-signer.service
+
+%preun
+%systemd_preun notary-server.service notary-signer.service
+
+%postun
+%systemd_postun_with_restart notary-server.service notary-signer.service
+
+%if 0%{?with_devel}
+%package devel
+Summary:       %{summary}
+BuildArch:     noarch
+
+%if 0%{?with_check} && ! 0%{?with_bundled}
+BuildRequires: golang(github.com/Sirupsen/logrus)
+BuildRequires: golang(github.com/Sirupsen/logrus/hooks/bugsnag)
+BuildRequires: golang(github.com/agl/ed25519)
+BuildRequires: golang(github.com/bugsnag/bugsnag-go)
+BuildRequires: golang(github.com/docker/distribution/context)
+BuildRequires: golang(github.com/docker/distribution/health)
+BuildRequires: golang(github.com/docker/distribution/registry/api/errcode)
+BuildRequires: golang(github.com/docker/distribution/registry/api/v2)
+BuildRequires: golang(github.com/docker/distribution/registry/auth)
+BuildRequires: golang(github.com/docker/distribution/registry/auth/silly)
```

Comment:
But %check does run the tests, so this `BuildRequires` seems appropriate.

(Well, maintaining all of the `!with_bundled` case in the spec file is dubious, but I think it is still useful to keep this flexibility to flip this on in the future.)